### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by pmkaboo**

### CODE REVIEW

**Comment by pmkaboo** [comment]
`- Add ability to print out the cheapest product and the most expensive product`
the methods price_min/max return an array of products with min/max prices, which means we can iterate through the array and call product.name, does this count as printing out the product, or a more elaborate solution is needed?

File: product.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Product
+	attr_accessor :name, :price
+
+	def initialize name, price
+		@name = name
+		@price = price.to_f
```
Comment (martincik) [inline product.rb]: Are you sure you want to represent price as float number?

File: products.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+require './product'
+
+class Products
+	attr_reader :collection
+
+	def initialize *products
+		@collection = products
```
Comment (martincik) [inline products.rb]: What if I pass in array of products instead of params?

File: products.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+		@collection << product
+		self
+	end
+
+	def price_sum
+		@collection.map(&:price).sum
+	end
+
+	def price_min
+		@collection.group_by(&:price).min.last
```
Comment (martincik) [inline products.rb]: 1) This should return a product with the minimum price. Your code is returning an array of just one product. Why?
2) Why not just use simple `Array#min_by`? Simple Google search would reveal for you the best and also correct solution. Why haven't you Google this?

File: products.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	end
+
+	def price_max
+		@collection.group_by(&:price).max.last
```
Comment (martincik) [inline products.rb]: ditto as min price

File: products.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+require './product'
+
+class Products
```
Comment (martincik) [inline products.rb]: Why not call this `Cart` or `Basket` as this whole app should be e-commerce simulation.

File: product.rb (hunk 1 above)
Comment (martincik) [reply product.rb]: What are the problems with Float numbers used for price?

File: products.rb (hunk 3 above)
Comment (pmkaboo) [reply products.rb]: yes, i know about min_by, but, it wouldnt work in situation where we have multiple products at min price.

File: products.rb (hunk 5 above)
Comment (pmkaboo) [reply products.rb]: thats a really good idea, most of my experience comes from working on internal systems for various companies, so i kinda have a hole in e-commerce naming conventions :D

File: products.rb (hunk 3 above)
Comment (martincik) [reply products.rb]: Good. Then you should have asked for input here as the spec states:
```
Add ability to compute the cheapest product and the most expensive product
```
and there's no plural. In real-world we need to communicate our intention before we do implement stuff.

File: products.rb (hunk 5 above)
Comment (martincik) [reply products.rb]: :-) So you're not using e-shops as a customer? ;)

File: products.rb (hunk 3 above)
Comment (pmkaboo) [reply products.rb]: yeah i tried to do that, but i cant imagine a scenario where you would want to lose information about the other cheapest products, so i didnt want to bother you for no reason.

File: products.rb (hunk 5 above)
Comment (pmkaboo) [reply products.rb]: only local ones, and they use 'kosik' :D

File: products.rb (hunk 2 above)
Comment (pmkaboo) [reply products.rb]: then it wont work :D, but im not sure if it would be my fault

**Pull request merged**